resolvers.py

```diff
@@ -19,11 +21,62 @@ def get_dates_and_price_check(
             **search_criteria
     ) -> List[DatesCheck]:
         validated_search_criteria = DatesAndPriceCheckInputValidator(**search_criteria)
-        results_list = []
+        houses_list = {}
         for listing_id in validated_search_criteria.listing_ids:
-            listing = data[listing_id]
-            dates_check = DatesCheck(
-                listing_id=listing.listing_id
-            )
-            results_list.append(dates_check)
+            houses_list[listing_id] = data[listing_id]
+        return QueryResolvers.getResultsList(validated_search_criteria, houses_list)
+
+    @staticmethod
+    @query.field("ranked_search_results")
+    def get_ranked_search_results(
+            _,
+            info,
+            **search_criteria
+    ) -> List[DatesCheck]:
+        validated_search_criteria = RankedSearchResultInputValidator(**search_criteria)
+        return QueryResolvers.getResultsList(validated_search_criteria, data)
+
+    @staticmethod
+    def getResultsList(validated_search_criteria, houses_list: dict) -> List[DatesCheck]:
+        results_list: List[DatesCheck] = []
+        checkin_Desired = validated_search_criteria.desired_checkin
+        checkout_Desired = validated_search_criteria.desired_checkout
+        if checkin_Desired is None and checkout_Desired is None:
+            for house in houses_list.values():
+                data_check = DatesCheck(house.listing_id, True, house.available_ranges[0].range_start,
```

Comment:
What happens if the house has no available ranges? Will you get an IndexError?

Also, same thing regarding naming. What is `data_check`?

Reply:
Went under the assumption that the data we are referring to, contains only listings with available ranges and that these are sorted by date. Not the case we would have when we are dealing with real scenario but wanted to simplify the use case for the toy app (so assumed all houses have available data and range.)

Ideally I would have an additional helper functions to retrieve/modify the ranges/dates instead of hardcoding a 0. Would probably fit better with the repository design pattern as we would be dealing with data from DB. Let me know if you want me to assume the real life scenarios for the mock data as well, as then I would add a few more cases and conditions.